### DISCUSSION TITLE
如何根据版本号添加包配置

### DISCUSSION THREAD

**Discussion by MasterBalll** [xmake-io/xmake#3477] [Q&A] [answered]
比如包定义里有这么一个配置，它是 v0.2 新增的，如何能让它只在使用 v0.2 版本的包时生效，对于使用更早版本的包无效，似乎在 on_load 中添加是不生效的。
目前是看到了这样的现象，就是添加这个新的配置后，那些使用旧版本包的工程也要被迫重新安装一遍带有这个新配置的包，所以想问问这种情况该怎么做。
```lua
package("abc")
    ...
    add_versions("0.2.0","xxx")
    add_versions("0.1.1","xxx")

    add_configs("sim", {description = "Set simulation.", default = "pc"})
    ...
```

**Comment by waruqi** [comment, marked as answer]
add_requires("abc 0.1.1") 手动定下版本 

或者走版本锁定 https://xmake.io/#/zh-cn/package/remote_package?id=%e4%be%9d%e8%b5%96%e5%8c%85%e7%9a%84%e9%94%81%e5%ae%9a%e5%92%8c%e5%8d%87%e7%ba%a7

**Reply by MasterBalll** [reply]
就是在说加了 `add_requires("abc 0.1.1")` 的工程阿，原先明明安装好了 v0.1.1 版本，但因为新加了 sim 的配置，就又需要重新安装一遍。
关联讨论 https://github.com/xmake-io/xmake/issues/3430 ，根据之前的讨论，因为有根据不同自定义选项/配置切换不同包（requires）的需求，目前的版本锁定又没法用。
那么只有重新安装一条路了嘛……

**Reply by waruqi** [reply]
因为你的 add_configs 设置了 default 。。所有版本都会默认带上这个  pc default。。配置变了，，buildhash 变了，就会重装。。

你可以仅仅只对 0.2 设置 default 。。也可以不设置 Default . on_install 里面，再去默认处理

**Reply by MasterBalll** [reply]
> 你可以仅仅只对 0.2 设置 default 。。

这个要咋做？
不过不设默认值，在 on_install 里做默认处理看起来可行。

**Reply by waruqi** [reply]
你就 on_install 里面搞下呗。。简单直接

**Reply by MasterBalll** [reply]
好吧，确实是可以的。只是这样就查不到（xmake q --info）这个根据配置判断的默认值了。

**Reply by waruqi** [reply]
这个没办法。配置默认值，就是等于告诉所有版本都生效这个默认配置。。原本就是要全版本重装的。。因为配置确实变了。。
